cpp/src/datetime/datetime_ops.cu

```diff
@@ -180,6 +191,101 @@ std::unique_ptr<column> apply_datetime_op(column_view const& column,
   return output;
 }
 
+struct add_calendrical_months_functor {
+  column_view timestamp_column;
+  column_view months_column;
+  mutable_column_view output;
+
+  add_calendrical_months_functor(column_view tsc, column_view mc, mutable_column_view out)
+    : timestamp_column(tsc), months_column(mc), output(out)
+  {
+  }
+
+  // std chrono implementation is copied here due to nvcc bug
```

Comment:
I faced same issue while trying to use `simt::std::chrono` in datetime code. 
https://github.com/rapidsai/cudf/blob/420a41027779f110b412a64d79749fb0621d387a/cpp/src/strings/convert/convert_datetime.cu#L260-L270

Reply:
What's the issue? The code doesn't tell me what the issue is.

Reply:
there are at least 3 versions of this function in cudf - one in this pr, one that @karthikeyann referenced and one in the cuio code. 

https://github.com/rapidsai/cudf/blob/08a69d21f346e25601b9efef15b2a8e48ff91f80/cpp/src/io/csv/datetime.cuh#L104

presumably they were added because the `std::chrono` functions couldn't be used on the device due to nvcc issue (bug number is documented in the code).